### ISSUE TITLE
Very imbalanced memory usage.

### ISSUE THREAD

**Issue by JiexingQi** [ElementAI/picard#15]
Hi， @tscholak . I'm fine-tuning the t5-base on 8 GeForce RTX 3090 GPUs(each memory is about 24GB). And I want to reproduce your experimet for t5-base with exact-setmatch accuracy (EM) of 57.2.
My train.json setting is :
```
{
    "run_name": "t5-spider",
    "model_name_or_path": "t5-base",
    "dataset": "spider",
    "source_prefix": "",
    "schema_serialization_type": "peteshaw",
    "schema_serialization_randomized": false,
    "schema_serialization_with_db_id": true,
    "schema_serialization_with_db_content": false,
    "normalize_query": true,
    "target_with_db_id": true,
    "output_dir": "/train",
    "cache_dir": "/transformers_cache",
    "do_train": true,
    "do_eval": true,
    "fp16": false,
    "num_train_epochs": 3072,
    "per_device_train_batch_size": 2,
    "per_device_eval_batch_size": 2,
    "gradient_accumulation_steps": 128,
    "label_smoothing_factor": 0.0,
    "learning_rate": 1e-4,
    "adafactor": true,
    "adam_eps": 1e-6,
    "lr_scheduler_type": "constant",
    "warmup_ratio": 0.0,
    "warmup_steps": 0,
    "seed": 1,
    "report_to": ["wandb"],
    "logging_strategy": "steps",
    "logging_first_step": true,
    "logging_steps": 4,
    "load_best_model_at_end": true,
    "metric_for_best_model": "exact_match",
    "greater_is_better": true,
    "save_total_limit": 128,
    "save_steps": 64,
    "evaluation_strategy": "steps",
    "eval_steps": 64,
    "predict_with_generate": true,
    "num_beams": 4,
    "num_beam_groups": 1,
    "use_picard": false
}
```
and the GPU usage is :
```
(base) jxqi@server2:~$ nvidia-smi
Tue Oct 26 11:17:16 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.57.02    Driver Version: 470.57.02    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0 Off |                  N/A |
| 55%   57C    P2   153W / 350W |   6194MiB / 24268MiB |     97%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce ...  On   | 00000000:25:00.0 Off |                  N/A |
| 57%   58C    P2   143W / 350W |   6192MiB / 24268MiB |     91%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA GeForce ...  On   | 00000000:41:00.0 Off |                  N/A |
| 52%   55C    P2   138W / 350W |   6192MiB / 24268MiB |     86%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA GeForce ...  On   | 00000000:61:00.0 Off |                  N/A |
| 49%   54C    P2   156W / 350W |   6194MiB / 24268MiB |     95%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   4  NVIDIA GeForce ...  On   | 00000000:81:00.0 Off |                  N/A |
| 54%   56C    P2   144W / 350W |   6192MiB / 24268MiB |     93%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   5  NVIDIA GeForce ...  On   | 00000000:A1:00.0 Off |                  N/A |
| 49%   54C    P2   142W / 350W |   6192MiB / 24268MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   6  NVIDIA GeForce ...  On   | 00000000:C1:00.0 Off |                  N/A |
| 53%   55C    P2   159W / 350W |   6192MiB / 24268MiB |     57%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   7  NVIDIA GeForce ...  On   | 00000000:E1:00.0 Off |                  N/A |
| 49%   54C    P2   143W / 350W |  23060MiB / 24268MiB |     92%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A    356248      C   python                           6189MiB |
|    1   N/A  N/A    356248      C   python                           6189MiB |
|    2   N/A  N/A    356248      C   python                           6189MiB |
|    3   N/A  N/A    356248      C   python                           6189MiB |
|    4   N/A  N/A    356248      C   python                           6189MiB |
|    5   N/A  N/A    356248      C   python                           6189MiB |
|    6   N/A  N/A    356248      C   python                           6189MiB |
|    7   N/A  N/A    356248      C   python                          23057MiB |
+-----------------------------------------------------------------------------+
```
it seems that it use Dataparallel mode(because it is only one process.), and the memory usage is very imbalanced and I have to  set `"per_device_train_batch_size": 2,`   to run it. 

Could you tell me how to  solve this problem? Thanky you @tscholak
